src/main/java/com/ericsson/gerrit/plugins/eiffel/messaging/EiffelEventSender.java

```diff
@@ -1,25 +1,121 @@
 
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.ericsson.eiffelcommons.utils.HttpRequest;
+import com.ericsson.eiffelcommons.utils.ResponseEntity;
 import com.ericsson.gerrit.plugins.eiffel.configuration.EiffelPluginConfiguration;
 import com.ericsson.gerrit.plugins.eiffel.events.EiffelEvent;
+import com.ericsson.gerrit.plugins.eiffel.exceptions.EiffelEventSenderException;
+import com.ericsson.eiffelcommons.utils.HttpRequest.HttpMethod;
 import com.google.gson.Gson;
-import com.google.gson.JsonObject;
 
 public class EiffelEventSender {
+    private static final String GENERATE_PUBLISH_ENDPOINT = "/generateAndPublish/";
+    private static final String MESSAGE_PROTOCOL = "mp";
+    private static final String MESSAGE_TYPE = "msgType";
+    private static final String EIFFEL_PROTOCOL = "eiffelsemantics";
+    private static final String CONTENT_TYPE_HEADER = "Content-Type";
+    private static final String APPLICATION_JSON = "application/json";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EiffelEventSender.class);
 
     private final Gson gson = new Gson();
-    private final EiffelPluginConfiguration pluginConfig;
+    private String eiffelMessage;
+    private String eiffelType;
+    private EiffelPluginConfiguration pluginConfig;
+    private HttpRequest httpRequest;
 
     public EiffelEventSender(EiffelPluginConfiguration pluginConfig) {
+        this.httpRequest = new HttpRequest(HttpMethod.POST);
         this.pluginConfig = pluginConfig;
     }
 
-    public void send(EiffelEvent eiffelEvent) {
-        JsonObject eiffelEventJson = gson.toJsonTree(eiffelEvent).getAsJsonObject();
-        System.out.println("Event to send::: " + eiffelEventJson.toString());
-        System.out.println("REMReM URL ::: " + pluginConfig.getRemremPublishURL());
-        // TODO Implement event sending using REMReM details from pluginconfig.
+    public EiffelEventSender(EiffelPluginConfiguration pluginConfig, HttpRequest httpRequest) {
+        this.httpRequest = httpRequest;
+        this.pluginConfig = pluginConfig;
+    }
+
+    /**
+     * Sends a REMReM eiffel message using
+     *
+     */
+    public void send() {
+        try {
+            generateAndPublish();
+        } catch (URISyntaxException | IOException | EiffelEventSenderException e) {
+            LOGGER.error("Failed to send eiffel message.", e);
+        }
+    }
+
+    public EiffelEventSender setMessage(final EiffelEvent eiffelMessage) {
+        this.eiffelMessage = gson.toJsonTree(eiffelMessage).toString();
+        return this;
+    }
+
+    public EiffelEventSender setType(final String eiffelType) {
+        this.eiffelType = eiffelType;
+        return this;
```

Comment:
If these parameters are mandatory. Why are they not part of the constructor?

Reply:
The way I had originally designed it was to send them as a data model since I didn't want too many parameters but again this was changed at the request of others to the way it is now. I understand your point of view and I even programmed it in a similar fashion first but once again this change it at odds with other peoples requests. What am I to do? Hahaha....

Reply:
This one I don't know what to do about, I think we will just leave it for now.